llvm/test/Transforms/LoopVectorize/AArch64/vectorize-atan2-darwin.ll

```diff
@@ -0,0 +1,43 @@
+; RUN: opt -vector-library=Darwin_libsystem_m -passes=inject-tli-mappings,loop-vectorize -S < %s | FileCheck %s
+
+target datalayout = "e-m:o-i64:64-i128:128-n32:64-S128"
```

Comment:
I don't know if adding a new test file is the right approach. 
Most of the other tests live in:
`llvm/test/Transforms/LoopVectorize/AArch64/veclib-calls-libsystem-darwin.ll`

Why would atan2 get its own file?

Reply:
Can we get atan2 added to isTriviallyVectorizable please?

Reply:
sure, I can address both of these.  maybe sometime next week.

Reply:
@tex3d  is working on a PR that would add atan2 to `isTriviallyVectorizable`. He might have some perspective on this PR.

Reply:
> I don't know if adding a new test file is the right approach.

fixed this by moving the test into an assertion inside `addVectorizableFunctions`. Now it's a bit harder to miss the entry in `hasOptimizedCodeGen` when adding new ones.

Reply:
> Can we get atan2 added to isTriviallyVectorizable please?

done

Reply:
I have a PR (#113637) adding atan2 to `isTriviallyVectorizable`, as well as adding missing massv cases for the intrinsic.  This was waiting behind other lowering PRs that have now merged.  I can rebase my PR on top of this one and make it about the massv change only once this merges.

I was trying to investigate what impact adding atan2 to `isTriviallyVectorizable` actually has, since all the supported veclib call vectorization occurs without it.  I can't seem to identify a test to modify or add which demonstrates/verifies the impact of the `isTriviallyVectorizable` change alone.

Reply:
The new tests in e73b49a3125964b0b4f35b4a5aab8d94bc5c7335 don't pass without the `isTriviallyVectorizable` change, for what that's worth.

Reply:
maybe that belongs in a different PR. do you want to pull it into yours?

Reply:
@jroelofs 
I'm ready to update my PR with the atan2 additions from here, if that's what you'd prefer (I have them tested and ready to push).  It will introduce a conflict if one merges first from the change in `hasOptimizedCodeGen`, but it should be easy to resolve.

If you'd like me to do that, you can revert those changes from this PR, but I'm still curious about what tests the impact of the `hasOptimizedCodeGen` changes in this PR.

Reply:
feel free to pick e73b49a3125964b0b4f35b4a5aab8d94bc5c7335 into your PR, and I'll revert it from mine, and then resurrect the test that covers the `hasOptimizedCodeGen` bit that I removed because of https://github.com/llvm/llvm-project/pull/109479#discussion_r1781856938

Reply:
The changes I pushed to my PR included adding atan2 cases to `hasOptimizedCodeGen` in `llvm/include/llvm/Analysis/TargetLibraryInfo.h`.  Should I revert that one change?  I'm currently testing without it to see if it has no measurable effect on my PR.

Reply:
you can keep them. now that that table is sorted, we shouldn't have much of a (or any?) merge conflict.

Reply:
https://github.com/llvm/llvm-project/pull/113637#pullrequestreview-2400460958

Reply:
Thanks!